src/config.ts

```diff
@@ -77,3 +87,89 @@ export const setConfig = (newConfig: ConfigOpts): Config => {
     ...newConfig,
   });
 };
+
+/**
+ * Get the provider and if undefined, throw.
+ * @param opts overrides for the current configuration.
+ */
+export const requireGetProvider = (opts?: ConfigOpts): ethers.providers.Provider => {
+  const c = getConfig(opts);
+  if (!c.provider) throw MissingProvider;
+  return c.provider;
+};
+
+/**
+ * Get the signer and if undefined, throw.
+ * @param opts overrides for the current configuration.
+ */
+export const requireGetSigner = (opts?: ConfigOpts): ethers.Signer => {
+  const c = getConfig(opts);
+  if (!c.signer) throw MissingSigner;
+  return c.signer;
+};
+
+/**
+ * Get the store and if undefined, throw.
+ * @param opts overrides for the current configuration.
+ */
+export const requireGetStore = (opts?: ConfigOpts): StoreInterface => {
+  const c = getConfig(opts);
+  if (!c.store) throw MissingStore;
+  return c.store;
+};
+
+/**
+ * Get the currentFromId and if undefined, throw.
+ * @param opts overrides for the current configuration.
+ */
+export const requireGetCurrentFromId = (opts?: ConfigOpts): string => {
```

Comment:
I like the distinction between getters that do and don't throw. Maybe it would better if the names only had either a `get` prefix or a `require` prefix? Like, this one could be `requireCurrentFromId` instead.

Reply:
I didn't add plain getters for everything, only because they weren't needed anywhere. I did not make everything 'require' because for queue and contracts, those are not optional fields and would never throw.

 This is not meant to be a final version.  The intent was to add non-throwing getters as needed. So if it's fine to have something undefined, then there are two options; use getConfig, or create a new getter.